python/ray/tests/test_autoscaling_policy.py

```diff
@@ -0,0 +1,579 @@
+        self.provider = MockProvider()
+        simulator = Simulator(config_path, self.provider)
+
+        start_count = 0
+
+        def start_callback():
+            nonlocal start_count
+            start_count += 1
+
+        tasks = [
+            Actor(
+                duration=float("inf"),
+                resources={"CPU": 1},
+                start_callback=start_callback,
+            )
+            for _ in range(5000)
+        ]
+        simulator.submit(tasks)
+
+        time = 0
+        while start_count < len(tasks):
+            time = simulator.step()
+
+        assert time < 200
+
+    def testManyPlacementGroups(self):
+        config = copy.deepcopy(SAMPLE_CLUSTER_CONFIG)
+        config_path = self.write_config(config)
+        self.provider = MockProvider()
+        simulator = Simulator(config_path, self.provider)
+
+        start_count = 0
+
+        def start_callback():
+            nonlocal start_count
+            start_count += 1
+
+        placement_group_requests = []
+
+        for _ in range(500):
+            placement_group_requests.append(
+                PlacementGroup(
+                    duration=float("inf"),
+                    bundles=[{"CPU": 1}, {"CPU": 2}],
+                    strategy=PlacementStrategy.STRICT_PACK,
+                    start_callback=start_callback,
+                )
+            )
+
+        for _ in range(500):
+            placement_group_requests.append(
+                PlacementGroup(
+                    duration=float("inf"),
+                    bundles=[{"CPU": 1}, {"CPU": 2}],
+                    strategy=PlacementStrategy.STRICT_SPREAD,
+                    start_callback=start_callback,
+                )
+            )
+
+        # SPREAD and PACK tests fail, but under the real GCS placement group
+        # scheduling algorithm we could also be left in a situation in which
+        # the autoscaler thinks the placement group is placeable, but the
+        # placement group scheduler doesn't know how to schedule it.
+
+        # for _ in range(500):
+        #     placement_group_requests.append(PlacementGroup(
+        #         duration=float("inf"), bundles=[{"CPU": 1}, {"CPU": 2}],
+        #         strategy=PlacementStrategy.PACK,
+        #         start_callback=start_callback))
+
+        # for _ in range(500):
+        #     placement_group_requests.append(PlacementGroup(
+        #         duration=float("inf"),
+        #         bundles=[{"CPU": 2}, {"CPU": 1}],
+        #         strategy=PlacementStrategy.SPREAD,
+        #         start_callback=start_callback))
+
+        simulator.submit(placement_group_requests)
+
+        time = 0
+        while start_count < len(placement_group_requests) and time < 500:
+            time = simulator.step()
+
+        # for ip, node in simulator.ip_to_nodes.items():
+        #     print(f"{ip}: {node.available_resources}, {node.total_resources}")
+        #     pass
+
+        assert time < 200
```

Comment:
Why this asserts 200 but in line 572 its 500?

Reply:
i think this has been fixed